examples/jsm/utils/CameraUtils.js

```diff
@@ -64,10 +66,33 @@ function frameCorners( camera, bottomLeftCorner, bottomRightCorner, topLeftCorne
 		camera.fov =
 			MathUtils.RAD2DEG / Math.min( 1.0, camera.aspect ) *
 			Math.atan( ( _vec.copy( pb ).sub( pa ).length() +
-							( _vec.copy( pc ).sub( pa ).length() ) ) / _va.length() );
+				( _vec.copy( pc ).sub( pa ).length() ) ) / _va.length() );
 
 	}
 
 }
 
-export { frameCorners };
+/** Calculate a PerspectiveCamera frustum's corners at a given distance from the camera.
+ * Returns an object with topLeft, topRight, bottomRight, and bottomLeft properties. Each property is a Vector3.
+ * Should work on all PerspectiveCameras where the frustum is rectangular in shape.
+ * Including asymmetric rectangular frustums like those used by webXR cameras.
+ * @param {number} distance - Distance from the camera
+ * @param {camera} camera - The PerspectiveCamera to calculate the frustum corners from */
+function getFrustumCorners( camera, distance ) {
+
+	camera.getBounds( distance, _minTarget, _maxTarget );
+
+	camera.updateMatrixWorld( true, false );
+
+	// .set() -> Calculates the corner position
+	// .applyMatrix4() -> Accounts for camera position/rotation/scale
+	return {
+		topLeft: _vec.set( _minTarget.x, _maxTarget.y, - distance ).applyMatrix4( camera.matrixWorld ).clone(),
+		topRight: _vec.set( _maxTarget.x, _maxTarget.y, - distance ).applyMatrix4( camera.matrixWorld ).clone(),
```

Comment:
I would return values in camera space - not world space. `topRight`, `topLeft`, have no meaning once the camera is rotated.

Reply:
As I said, `topRight`, `topLeft`, have no meaning in world space. Return the values in camera space. Any user using this method knows how to convert a point from camera space into world space -- if that is what they want.

Reply:
Does topLeft fully loose all meaning if the camera is rotated? Agnostic of camera rotation or reference coordinate space, the topLeft vector will still be the point that is rendered to the topLeft of the screen.

The utility that seems most useful to add is the world-space representation so that you can easily use output vectors relative to other objects in the world-space.

Reply:
In addition to my previous comments, I do not think `.getFrustumBounds()` and `.getFrustumCorners()` should return results in different coordinate frames.

And I agree, `.getFrustumCorners()` provides little added utility.

Reply:
@Bug-Reaper To avoid an impasse here, I suggest you remove `getFrustumCorners()` from this PR and create a separate one. In this way, we can at least get the changes to `PerspectiveCamera` get merged now.

Reply:
Okay guys, gonna remove `getFrustumCorners()` from this PR so we can get the goods merged ⭐️

I do believe there's tangible utility added from a method that returns the corners in world-space. Perhaps the name conventions can be improved to clearly indicate the functionality. Will take the feedback from this discussion to heart and try to keep it in mind when I get a chance to do a seperate PR for this method.